tests/src/com.github.ben-manes.caffeine/caffeine/2.9.3/src/test/java/com_github_ben_manes_caffeine/caffeine/CaffeineTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_github_ben_manes_caffeine.caffeine;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.CaffeineSpec;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.github.benmanes.caffeine.cache.Scheduler;
+import com.google.common.testing.FakeTicker;
+import org.junit.jupiter.api.Test;
+
+import java.lang.ref.Cleaner;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CaffeineTest {
+    @Test
+    void testRefresh() {
+        LoadingCache<String, String> graphs = Caffeine.newBuilder().maximumSize(10_000).refreshAfterWrite(1, TimeUnit.MINUTES)
+                .build(key -> key.equals("Hello") ? "World" : "Universe");
+        assertThat(graphs.get("Hello")).isEqualTo("World");
+        assertThat(graphs.getAll(List.of("Hi", "Aloha"))).isEqualTo(Map.of("Hi", "Universe", "Aloha", "Universe"));
+    }
+
+    @Test
+    void testStatistics() {
+        Cache<String, String> graphs = Caffeine.newBuilder().maximumSize(10_000).recordStats().build();
```

Comment:
Why are you using `recordStats` here? When are you queering it?  Also, which metadata corresponds to this test? I am asking this because it looks like this test tests same thing as some other tests, and therefore maybe can be removed. Is that correct?

Reply:
- This seems to be a wrong local refactoring. When verifying the different performances of Guava Cache and Caffeine under Native Image, I reused a set of temporary logic.
- `testStatistics()` has added additional assertions.